### DISCUSSION TITLE
HmIP-WKP Keypad events

### DISCUSSION THREAD

**Discussion by EddyVanEm** [SukramJ/aiohomematic#1253] [Q&A] [answered]
Hi everyone,
I have a new device - the HmIP-WKP. Other than opening the door with this keypad I would like to add some automations if somone opens the door with the correct PIN on the keypad. Unfortunately I don't see any trigger or whatsoever that I can use for an automation. 

Any help appreciated!

**Comment by SukramJ** [comment, marked as answer]
There should be multiple device triggers and events displayed for the device.
To make it work you need to follow [these](https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices) instructions.

**Reply by romeocontrol** [reply]
[these](https://github.com/danielperna84/custom_homematic#events-for-homematicip-devices) instructions
made my WKP working.

> With RaspberryMatic no program is needed for buttons. Events can directly activated/deactivated within ->Settings->Devices.

This hint would be a bit faster to execute, but does not work for the WKP-Keypad, an Activate button is noch shown.
